src/test/java/core/basesyntax/service/ActionStrategyImplTest.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.action.ActionHandler;
+import core.basesyntax.service.action.BalanceAction;
+import core.basesyntax.service.action.PurchaseAction;
+import core.basesyntax.service.action.ReturnAction;
+import core.basesyntax.service.action.SupplyAction;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class ActionStrategyImplTest {
```

Comment:
add a negative test

Reply:
Here we can't add notOk because we check this before